# Hypothesis Testing

## Problem statement 1: 
A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

Minitab File : Cutlets.mtw

In [103]:
# importing libraries
import pandas as pd
import numpy as np
from scipy import stats

In [104]:
ls

 Volume in drive C is program files
 Volume Serial Number is 5C10-FCA7

 Directory of C:\Users\Sayali Hatwar\OneDrive\Desktop\Assignment files\Hypothesis assignment

13-01-2023  02:11    <DIR>          .
13-01-2023  02:11    <DIR>          ..
13-01-2023  00:25    <DIR>          .ipynb_checkpoints
13-01-2023  02:11            33,903 Assignment no. 3.ipynb
12-12-2022  16:57                87 BuyerRatio.csv
12-12-2022  16:57            13,423 Costomer+OrderForm.csv
12-12-2022  16:57               525 Cutlets.csv
13-01-2023  00:26            61,688 Hypothesis_Testing_Assignment.pptx
12-12-2022  16:57             3,487 LabTAT.csv
               6 File(s)        113,113 bytes
               3 Dir(s)  125,546,184,704 bytes free


In [105]:
# Reading data file
cutlet = pd.read_csv('cutlets.csv')
cutlet.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

# EDA

In [106]:
# description of data
cutlet.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [107]:
# Checking the information
cutlet.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35 entries, 0 to 34
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Unit A  35 non-null     float64
 1   Unit B  35 non-null     float64
dtypes: float64(2)
memory usage: 688.0 bytes


In [108]:
# checking the duplicates in data
cutlet.duplicated().sum()

0

In [109]:
# Counting unit A and unit B
cutlet.value_counts()

Unit A  Unit B
6.4376  7.5093    1
7.2828  7.3070    1
7.0852  7.5459    1
7.1560  7.4220    1
7.1561  6.6965    1
7.2163  7.0133    1
7.2705  7.4314    1
7.2783  7.1180    1
7.2854  7.1688    1
6.9952  6.7594    1
7.3012  6.7878    1
7.3495  6.7478    1
7.3871  6.8110    1
7.3930  6.8810    1
7.3943  6.5780    1
7.4488  7.1522    1
7.0621  6.6606    1
6.9438  6.8889    1
6.5341  6.5217    1
6.8236  7.0503    1
6.5797  7.1581    1
6.5970  6.6672    1
6.6801  6.9182    1
6.6840  7.2402    1
6.7794  7.0992    1
6.8090  6.7703    1
6.8394  7.0240    1
6.9431  6.3346    1
6.8568  6.9399    1
6.8755  7.2212    1
6.9157  6.7300    1
6.9246  6.7652    1
6.9256  6.0380    1
6.9405  7.3875    1
7.5169  7.4059    1
dtype: int64

### Two - sample test

In [110]:
# Providing names to each unit
X = cutlet['Unit A']
Y = cutlet['Unit B']

In [111]:
# Calculating pvalue and test statistics
p_value = stats.ttest_ind(X,Y,axis=0,alternative='two-sided')
p_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.47223947245995)

In [112]:
p_value[1]

0.47223947245995

In [113]:
# Inference
if p_value[1]<0.05:
    print('H0 is rejected and there is difference in size of cutlet')
else:
    print('H0 is accepted and there is no difference in size of cutlet')

H0 is accepted and there is no difference in size of cutlet


## Inference

Ho -> 0.4722

H1 -> 1-0.4722 = 0.5278 

alpha = 0.05

t- statistics = 0.722

As H0 is accepted there is no significance difference in size of cutlet

## Problem Statement 2:
    
  A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories.TAT is defined as sample collected to report dispatch.
   
  Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.
 
 
    Minitab File: LabTAT.mtw


In [114]:
# importing libraries
import pandas as pd
import numpy as np
from scipy import stats

In [115]:
ls

 Volume in drive C is program files
 Volume Serial Number is 5C10-FCA7

 Directory of C:\Users\Sayali Hatwar\OneDrive\Desktop\Assignment files\Hypothesis assignment

13-01-2023  02:11    <DIR>          .
13-01-2023  02:11    <DIR>          ..
13-01-2023  00:25    <DIR>          .ipynb_checkpoints
13-01-2023  02:11            33,903 Assignment no. 3.ipynb
12-12-2022  16:57                87 BuyerRatio.csv
12-12-2022  16:57            13,423 Costomer+OrderForm.csv
12-12-2022  16:57               525 Cutlets.csv
13-01-2023  00:26            61,688 Hypothesis_Testing_Assignment.pptx
12-12-2022  16:57             3,487 LabTAT.csv
               6 File(s)        113,113 bytes
               3 Dir(s)  125,546,184,704 bytes free


In [116]:
# Reading csv file
TAT = pd.read_csv('LabTAT.csv')
TAT.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [117]:
# Alloting name to experiment list
Z1 = TAT['Laboratory 1']
Z2 = TAT['Laboratory 1']
Z3 = TAT['Laboratory 1']
Z4 = TAT['Laboratory 1']

Now, we set up the hypothis

H0 : Z1 = Z2 = Z3 = Z4

H1 : Z1 != Z2 != Z3 != Z4    

In [118]:
# Calculating p_value
p_value = stats.f_oneway(TAT.iloc[:,0],TAT.iloc[:,1],TAT.iloc[:,2],TAT.iloc[:,3])
p_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

In [119]:
p_value[1]

2.1156708949992414e-57

In [120]:
# Inference
if p_value[1]<0.5:
    print('H0 is rejected and there is some difference in average TAT among the different laboratories')
else:
    print('H0 is accepted and there is no difference in average TAT among the different laboratories')

H0 is rejected and there is some difference in average TAT among the different laboratories


## Problem Statement 3:
    
  Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions
 
  Minitab File: Buyer Ratio.mtw


In [121]:
# Importing libraries
import pandas as pd
from scipy import stats
import numpy as np
from scipy.stats import chi2_contingency,chisquare

In [122]:
ls

 Volume in drive C is program files
 Volume Serial Number is 5C10-FCA7

 Directory of C:\Users\Sayali Hatwar\OneDrive\Desktop\Assignment files\Hypothesis assignment

13-01-2023  02:11    <DIR>          .
13-01-2023  02:11    <DIR>          ..
13-01-2023  00:25    <DIR>          .ipynb_checkpoints
13-01-2023  02:11            33,903 Assignment no. 3.ipynb
12-12-2022  16:57                87 BuyerRatio.csv
12-12-2022  16:57            13,423 Costomer+OrderForm.csv
12-12-2022  16:57               525 Cutlets.csv
13-01-2023  00:26            61,688 Hypothesis_Testing_Assignment.pptx
12-12-2022  16:57             3,487 LabTAT.csv
               6 File(s)        113,113 bytes
               3 Dir(s)  125,546,184,704 bytes free


In [123]:
# Reading the file
BR = pd.read_csv('BuyerRatio.csv')
BR.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

Now, we set the hypothis

H0 : Male = Female

H1 : Male != Female    

In [124]:
# converting to array
df = np.array([[50,142,131,70],[435,1523,1356,750]])
df

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [125]:
# Calculating contigency table
chi2_contingency(df)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [126]:
# Calculating p_value
p_value = chi2_contingency(df)[1]
p_value

0.6603094907091882

In [127]:
# Inference
if p_value<0.05:
    print('H0 is rejected and there is significant difference between male and female buyers ratio across region')
else:
    print('H0 is accepted and there is no significant difference between male and female buyers ration across region')

H0 is accepted and there is no significant difference between male and female buyers ration across region


## Problem Statement 4:
    
   Telecall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

Minitab File: CustomerOrderForm.mtw
 


In [128]:
# Importing libraries
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import chi2_contingency,chisquare

In [129]:
ls

 Volume in drive C is program files
 Volume Serial Number is 5C10-FCA7

 Directory of C:\Users\Sayali Hatwar\OneDrive\Desktop\Assignment files\Hypothesis assignment

13-01-2023  02:11    <DIR>          .
13-01-2023  02:11    <DIR>          ..
13-01-2023  00:25    <DIR>          .ipynb_checkpoints
13-01-2023  02:11            33,903 Assignment no. 3.ipynb
12-12-2022  16:57                87 BuyerRatio.csv
12-12-2022  16:57            13,423 Costomer+OrderForm.csv
12-12-2022  16:57               525 Cutlets.csv
13-01-2023  00:26            61,688 Hypothesis_Testing_Assignment.pptx
12-12-2022  16:57             3,487 LabTAT.csv
               6 File(s)        113,113 bytes
               3 Dir(s)  125,546,180,608 bytes free


In [130]:
# Reading the csv file
COF = pd.read_csv('Costomer+OrderForm.csv')
COF.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

Now, we set the hypothesis

H0 : NO defective % varies by centre

H1 : Defective % varies by centre    

In [131]:
# Reading the information 
COF.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Phillippines  300 non-null    object
 1   Indonesia     300 non-null    object
 2   Malta         300 non-null    object
 3   India         300 non-null    object
dtypes: object(4)
memory usage: 9.5+ KB


In [132]:
# Counting error free and defective values
print(COF['Phillippines'].value_counts())
print(COF['Indonesia'].value_counts())
print(COF['Malta'].value_counts())
print(COF['India'].value_counts())

Error Free    271
Defective      29
Name: Phillippines, dtype: int64
Error Free    267
Defective      33
Name: Indonesia, dtype: int64
Error Free    269
Defective      31
Name: Malta, dtype: int64
Error Free    280
Defective      20
Name: India, dtype: int64


In [133]:
# converting to array
df = np.array([[271,267,269,280],[29,33,31,20]])
df

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [134]:
# Calculating contingency table
chi2_contingency(df)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [135]:
# Calculating p_value from chi2 contigency table
p_value = chi2_contingency(df)[1]
p_value

0.2771020991233135

In [136]:
# Inference
if p_value < 0.05:
    print('H0 is rejected and there are some defective % varies by centre')
else:
    print('H0 is accepted and there are no defective % varies by centre')

H0 is accepted and there are no defective % varies by centre
